### ISSUE TITLE
Training for Ego4D Narration Stream

### ISSUE THREAD

**Issue by XiangTodayEatsWhat** [showlab/videollm-online#20]
Hello! Thanks for your great work!

The script for training streaming chat (in scripts/ego4d/live1+.sh) contains two parts of data: goalstep and narration.

When I reproduced it, I found that goalstep only contained 851 videos, while narration corresponded to 7466. 

I would like to ask if I can't train for narration stream if I only download goalstep? 
How should I download the data corresponding to narration? 
And is there a corresponding quantitative results for goalstep + narration?

Thank you very much!

**Comment by chenjoya** [comment]
Hello, the script for only narration is at https://github.com/showlab/videollm-online/blob/main/scripts/ego4d/narration/live1%2B.sh

The total is 9819 videos. Seems that you missed some videos while downloading Ego4D?

Let me explain about goalstep + narration. If you only train narration data, the model can only do narration rather than free-form dialogue (e.g., ask some question intermediately). So, we use goalstep to synthesis free-form dialogue data. However, the quantity of goalstep is not so enough. Thus we train narration + goalstep. 

The paper experiments are conducted on narration data only since it is not easy to evaluate free-form dialogue.

**Comment by XiangTodayEatsWhat** [comment]
Thank you so much for such a quick and clear reply! 
Because I originally referred to #14 and downloaded only the goalstep subset, now I understand!

**Comment by chenjoya** [comment]
Thanks! Feel free to contact me for any problems!

**Comment by wenyuqing** [comment]
Hello, I want to ask that what json files should I use to train the final model? If I want to train narration + goalstep to reproduce your results, shouls I use these 2 json files provided in [https://huggingface.co/datasets/chenjoya/videollm-online-chat-ego4d-134k](url) ?
videollm-online-chat-ego4d-134k/goalstep_livechat_trainval_filtered_21k.json  for goalstep and
/data/yq_code/videollm-online/videollm-online-chat-ego4d-134k/refined_narration_stream_train.json for narration
Is that correct?
Also, I'm a little confused about the configuration:
`--train_datasets ego4d_narration_stream_train`
and `--train_datasets  ego4d_refined_narration_stream_train`,
What's the difference of these two datasets and what should I use for training?

**Comment by wenyuqing** [comment]
So what should I use, the refined one or not refined one?

**Comment by chenjoya** [comment]
Use the refined one. 'C is doing sth ...' is weird for LLM to learn (in my opinion)

**Comment by wenyuqing** [comment]
Thanks!

**Comment by chenjoya** [comment]
No problem!

**Comment by XiangTodayEatsWhat** [comment]
Hello, I would like to ask if cli.py could be used for free-form conversations such as goalstep? 
When I used goalstep_livechat_trainval_filtered_21k.json for testing, I found it difficult to get real-time results.
For example:
<img width="1417" alt="image" src="https://github.com/user-attachments/assets/14a7a7e1-fc96-4376-af65-a2d49981b64e">
After entering the question at the corresponding time for the above item:
<img width="776" alt="image" src="https://github.com/user-attachments/assets/5a451197-cae0-4ae3-a620-0614cba979d6">
the result is as follows:
<img width="937" alt="image" src="https://github.com/user-attachments/assets/b962945e-c8f1-4157-87cd-af09245cbf2a">
Only one answer can be given after each question. 
Is it that the code logic of cli.py needs to be adjusted or is it difficult for the model itself to obtain the true value effect?

Thank you very much!
